src/lib.rs

```diff
@@ -644,6 +645,40 @@ where
     }
 }
 
+/// A panic payload indicating that a salsa revision was canceled.
+#[derive(Debug)]
+pub struct Canceled {
+    _private: (),
+}
+
+impl Canceled {
+    fn throw() -> ! {
+        std::panic::resume_unwind(Box::new(Self { _private: () }));
+    }
+}
+
+impl std::fmt::Display for Canceled {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("canceled")
+    }
+}
+
+impl std::error::Error for Canceled {}
```

Comment:
I think we should have this, but this was not obvious to me, so let me write down justification. 

I would expect Cancellation to not be treated as an error. Cancellation is a serendipitous success. It would be wrong to propagate Cancelled as Error. However, in salsa itself `Cancelled` is not used throughout the API, as we use unwinding, its used only at the boundady. For boundaries, I think treating it as an Error is probably fine